### PR TITLE
Created more account and accountclaim conditions

### PR DESCRIPTION
**Pull request by ArielLima**
Each existing code path that leads to a failed account has a new, representative condition with a descriptive TYPE, REASON and MESSAGE.

Card -> https://issues.redhat.com/browse/OSD-5396

### CODE REVIEW

File: pkg/apis/aws/v1alpha1/account_types.go [hunk 1]
```diff
@@ -89,6 +89,8 @@ const (
 	AccountReady AccountConditionType = "Ready"
 	// AccountFailed is set when account creation has failed
 	AccountFailed AccountConditionType = "Failed"
+	// AccountCreationFailed is set when an error occured during account creations
```
Comment (clcollins) [inline pkg/apis/aws/v1alpha1/account_types.go:92]: Would you mind making this "...during AWS account creation" to specify that it was a "real" account creation that failed, and not our operator CR object that failed?  Just to be clear what the difference is until the "AccountFailed" condition goes away.

**Review by clcollins — CHANGES_REQUESTED** [review]

**Comment by rogbas** [comment]
@ArielLima there is some lint and build errors, could you please check it? thanks

**Pull request merged**